File: sentimental/readability/readability.py
import cs50
import math


letters = 0
words = 1
sentences = 0
text = cs50.get_string("Text: ")

for char in text:
    if char.isalpha():
        letters += 1
    elif char == " ":
        words += 1
    elif char in [".", "!", "?"]:
        sentences += 1

index = 0.0588 * (letters / (words / 100)) - 0.296 * (sentences / (words / 100)) - 15.8
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {math.floor(index + 0.5)}")

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Word count is 1 higher than space count
    int letters = 0;
    int words = 1;
    int sentences = 0;
    string text = get_string("Text: ");

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        else if (text[i] == ' ')
        {
            words += 1;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }

    float index = 0.0588 * ((float) letters / ((float) words / 100)) - 0.296 * ((float) sentences / ((float) words / 100)) - 15.8;
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) (index + 0.5));
    }
}